lib/internal/modules/esm/fetch_module.js

```diff
@@ -127,7 +131,12 @@ function fetchWithRedirects(parsed) {
         err.message = `Cannot find module '${parsed.href}', HTTP 404`;
         throw err;
       }
-      if (res.statusCode > 303 || res.statusCode < 200) {
+      if (
+        (
+          res.statusCode > 303 &&
+          res.statusCode !== 307 &&
+          res.statusCode !== 308
+        ) || res.statusCode < 200) {
```

Comment:
This part makes me wonder if original flow is correct. If the response `isRedirect && !hasLocation`, shouldn't we throw? Especially if we deal with [`300 Multiple Choices`](https://httpwg.org/specs/rfc7231.html#status.300).
This concern is not blocking and can be addressed separately, the change itself LGTM.

Reply:
To be fair, I'm not even sure if there are reasonable use cases to import a body of response with any `2xx` code other than `200`.

Reply:
From a `GET` request?

Reply:
oh true, i wouldn't expect a 201 from a GET request :-) a 204 (no body) or 206 (partial body) maybe, though

Reply:
`206` probably shouldn't happen there (i.e. we don't send any `Range` so partial body is not what should be expected), `204` restricts having a body at all... Although yes, technically we can `import {} from 'https://example.com/204-no-content.mjs';` and there's nothing really wrong about that. :smile: